core/src/create.jl

```diff
@@ -201,6 +201,17 @@ end
 const nonconservative_nodetypes =
     Set{String}(["Basin", "LevelBoundary", "FlowBoundary", "Terminal", "User"])
 
+function get_chunk_sizes(config::Config, chunk_size::Int)::Vector{Int}
+    chunk_sizes = Int[]
+    if config.solver.autodiff
+        push!(chunk_sizes, chunk_size)
+    end
+    if config.solver.algorithm in ["Rodas5", "Rosenbrock23"]
+        push!(chunk_sizes, 1)
+    end
```

Comment:
Should the second `if` not be inside the first `if`? We only use autodiff when `config.solve.autodiff` right?

We might add other Rosenbrock options in the future, perhaps this check is a bit more robust:

`algorithm(config.solver) isa OrdinaryDiffEqRosenbrockAdaptiveAlgorithm`

if you do `using OrdinaryDiffEq: OrdinaryDiffEqRosenbrockAdaptiveAlgorithm`.

Reply:
Ah nice, I couldn't find that more robust check.

Reply:
I find this function convenient for finding the right abstract type:

```
julia> supertypes(Int)
(Int64, Signed, Integer, Real, Number, Any)
```